core/queryresultio/api/src/main/java/org/eclipse/rdf4j/query/resultio/QueryResultIO.java

```diff
@@ -316,24 +320,77 @@ public static TupleQueryResult parseTuple(InputStream in, QueryResultFormat form
 		throws IOException, QueryResultParseException, TupleQueryResultHandlerException,
 		UnsupportedQueryResultFormatException
 	{
-		TupleQueryResultParser parser = createTupleParser(format);
+		return parseTupleInternal(in, format, false);
+	}
+
+	/**
+	 * Parses a query result document and returns it as a TupleQueryResult object, with parsing done on a
+	 * separate thread in the background.<br>
+	 * IMPORTANT: As this method will spawn a new thread in the background, it is vitally important that it be
```

Comment:
Same as before: change "it" to "the TupleQueryResult" and move/copy this warning to the `@return` javadoc.